website/pages/docs/concepts/seal.mdx

```diff
@@ -84,80 +84,129 @@ to decrypt the master key at startup and unseal automatically.
+
 ## Seal Migration
 
-The seal can be migrated from Shamir keys to Auto Unseal, Auto Unseal to Shamir
-keys and Auto Unseal to another Auto Unseal.
+The seal can be migrated from Shamir Seal to KMS Seal, KMS Seal to Shamir Seal,
+and KMS Seal to another KMS Seal.
+
+~> **NOTE**: Seal migration process cannot be performed without downtime. Due to
+the technical underpinnings of the seal implementations, it is at this point not
+possible to perform seal migration without briefly bringing the whole cluster
+down. We understand that it can be hard for many deployments to face downtime,
+but we believe that switching seals is a rare event and hence we hope for the
+downtime to be considered as an acceptable trade-off.
+
+~> **NOTE**: Seal migration operation will require both old and new seals to be
+available during the migration. For example, migration from KMS seal to Shamir
+seal will require that the KMS key be accessible during the migration.
+
+~> **NOTE**: Seal migration from KMS seal to Shamir seal is not currently
+supported when using Vault Enterprise. We plan to support this officially in a
+future release.
+
+~> **NOTE**: Seal migration from KMS seal to KMS seal of same kind is not
+currently supported when using Vault Enterprise. We plan to support this
+officially in a future release.
+
+### Migration post Vault 1.4.0
 
-~> **NOTE**: The migration operation will require both seals to be available
-during the migration. For example, a migration from a cloud KMS seal to
-Shamir will require that the cloud KMS be accessible during the migration.
+These steps are common for seal migrations between any supported kinds and for
+any storage backend.
 
-### Migration From Shamir to Auto Unseal
+1. Take the standby nodes down, one node at a time and update the [seal
+configuration](/docs/configuration/seal). If the migration is from Shamir seal
+to KMS seal, add the desired new KMS seal block to the config. If the migration
+is from KMS seal to Shamir seal, add `disabled = "true"` to the old seal block.
+If the migration is from KMS seal to another KMS seal, add `disabled = "true"`
+to the old seal block and add the desired new KMS seal block.
 
-To migrate from Shamir keys to Auto Unseal, take your server cluster offline and update
-the [seal configuration](/docs/configuration/seal) with the appropriate seal
-configuration. Bring your server back up and leave the rest of the nodes offline if
-using multi-server mode, then run the unseal process with the `-migrate` flag and bring
-the rest of the cluster online.
+2. Bring the standby nodes back up one at a time and run the unseal command on
+each by supplying the `-migrate` flag. Supply Shamir unseal keys if old seal was
+Shamir, which will be migrated as the recovery keys for the KMS seal. Supply
+recovery keys if the old seal is one of KMS seals, which will be migrated as the
+recovery keys of the new KMS seal or as Shamir unseal keys if the new seal is
+Shamir.
```

Comment:
We should make it clear that step 1 & 2 should be done per node. e.g. Do step 1&2 on standby 1 then Do step 1&2 on standby 2, etc. Otherwise if you do step 1 on standby 1&2 then step 2 on standby 1&2 you've likely broken raft quorum.

Reply:
Merged step 1&2 as a single block of action and updated step 2 to have step 1 in a modular way. Let me know what you think of that.